gothic/gothic.go

```diff
@@ -99,6 +99,14 @@ var SetState = func(req *http.Request) string {
 // This is used to prevent CSRF attacks, see
 // http://tools.ietf.org/html/rfc6749#section-10.12
 var GetState = func(req *http.Request) string {
+
+	params := req.URL.Query()
+	if params.Encode() == "" && req.Method == "POST" {
+		req.ParseForm()
```

Comment:
Hey @wunderkind2k1, I just realized we actually stumbled upon the same issue. I opened https://github.com/markbates/goth/pull/357 with my proposal for the fix, but yours is more appropriate as it handles it in the `GetState()` function.

May I suggest to avoid calling `ParseForm()` here and use `req.FormValues("state")`? FormValues calls ParseForm internally, so there's not much difference, but the code will be shorter. What about doing something like this instead?

```go
var GetState = func(req *http.Request) string {
	if req.Form.Encode() == "" && req.Method == http.MethodPost {
                return req.FormValues("state")
	}

	return req.URL.Query().Get("state")
}
```

Reply:
Good idea. Thank you. 9a337777e3823df0d799d3013f8b9b163dc41480 should do it.

Reply:
Hey @macuenca - I have made your suggested changes. Maybe you are so kind and can close your pull request #357 then please?

Reply:
https://github.com/markbates/goth/pull/356/commits/1ace300f3fc29e2458b7d7ec3ead4d33efa86968 should do it. Thanks for pointing me to it @macuenca